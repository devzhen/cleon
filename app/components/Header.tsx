"use client";

import styled from "styled-components";

export default function Header() {
  return (
    <div>
      <HeaderTop />
      <HeaderBottom>
        <HeaderLogo>
          <HeaderColumn />
          <HeaderLogoTitleTop>
            <span>cl√àon</span>
          </HeaderLogoTitleTop>
          <HeaderLogoTitleBottom>
            <span>l`usine incontournable pour l`automobile de demain</span>
          </HeaderLogoTitleBottom>
        </HeaderLogo>
      </HeaderBottom>
    </div>
  );
}

const HeaderTop = styled.div`
  height: 4px;
  background-color: ${(props) => props.theme.colors.headerTopBackground};
  width: 100%;
`;

const HeaderBottom = styled.div`
  height: 65px;
  background-color: ${(props) => props.theme.colors.mainBackground};
  width: 100%;
`;

const HeaderLogo = styled.div`
  width: 600px;
  height: 57px;
`;

const HeaderColumn = styled.div`
  width: 5px;
  height: 44px;
  background-color: ${(props) => props.theme.colors.white};
  position: relative;
  top: 7px;
  left: 19px;
`;

const HeaderLogoTitleTop = styled.div`
  position: relative;
  top: -39px;
  left: 39px;

  span {
    font-size: 18px;
    color: ${(props) => props.theme.colors.white};
    text-transform: uppercase;
  }
`;

const HeaderLogoTitleBottom = styled.div`
  position: relative;
  top: -34px;
  left: 40px;

  span {
    font-size: 18px;
    color: ${(props) => props.theme.colors.headerLogoTitle};
    text-transform: uppercase;
  }
`;
