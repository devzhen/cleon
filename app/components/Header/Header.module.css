.header {
  --color-header-top-background: #363636;
  --color-header-logo-title: #232323;

  grid-column: 2 span;
  height: 66px;
  width: 100%;
}

.header > div:nth-child(1) {
  height: 4px;
  background-color: var(--color-header-top-background);
  width: 100%;
}

.header > div:nth-child(2) {
  height: 65px;
  background-color: var(--color-main-background);
  width: 100%;
}

.header-logo {
  width: 600px;
  height: 57px;
}

.header-logo > div:nth-child(1) {
  width: 5px;
  height: 44px;
  background-color: #ffffff;
  position: relative;
  top: 7px;
  left: 19px;
}

.header-logo-title {
  position: relative;
  top: -39px;
  left: 39px;
}

.header-logo-title span {
  font-size: 18px;
  color: #ffffff;
  text-transform: uppercase;
}

.header-logo-subtitle {
  position: relative;
  top: -34px;
  left: 40px;
}

.header-logo-subtitle span {
  font-size: 18px;
  color: var(--color-header-logo-title);
  text-transform: uppercase;
}
