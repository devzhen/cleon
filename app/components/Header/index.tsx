import styles from './Header.module.css';

export default function Header() {
  return (
    <header className={styles.header}>
      <div />
      <div>
        <div className={styles['header-logo']}>
          <div />
          <div className={styles['header-logo-title']}>
            <span>cl√àon</span>
          </div>
          <div className={styles['header-logo-subtitle']}>
            <span>l`usine incontournable pour l`automobile de demain</span>
          </div>
        </div>
      </div>
    </header>
  );
}
