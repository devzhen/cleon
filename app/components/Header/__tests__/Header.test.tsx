import { render } from '@testing-library/react';

import Header from '..';

describe('Header tests', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
