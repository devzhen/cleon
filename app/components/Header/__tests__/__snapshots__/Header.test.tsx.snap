// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests matches the snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div />
    <div>
      <div
        class="header-logo"
      >
        <div />
        <div
          class="header-logo-title"
        >
          <span>
            cl√àon
          </span>
        </div>
        <div
          class="header-logo-subtitle"
        >
          <span>
            l\`usine incontournable pour l\`automobile de demain
          </span>
        </div>
      </div>
    </div>
  </header>
</div>
`;
